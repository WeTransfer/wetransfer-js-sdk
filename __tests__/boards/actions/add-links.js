const routes = require('../../../src/config/routes');
const addLinksAction = require('../../../src/boards/actions/add-links');
const { RemoteBoard } = require('../../../src/boards/models');

describe('Add links action', () => {
  let board = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue([
      {
        id: 'random-hash',
        url: 'https://wetransfer.com',
        meta: {
          title: 'WeTransfer',
        },
        type: 'link',
      },
    ]);

    addLinks = addLinksAction({
      routes,
      request: mocks.request,
    });

    board = new RemoteBoard({
      id: 'board-id',
      items: [],
    });
  });

  it('should create an add links request', async () => {
    await addLinks(board, [
      {
        url: 'https://wetransfer.com',
        title: 'WeTransfer',
      },
    ]);
    expect(mocks.request.send).toHaveBeenCalledWith(
      {
        url: '/v2/boards/board-id/links',
        method: 'post',
      },
      [
        {
          url: 'https://wetransfer.com',
          title: 'WeTransfer',
        },
      ]
    );
  });

  it('should return a RemoteLink object', async () => {
    const links = await addLinks(board, [
      {
        url: 'https://wetransfer.com',
        title: 'WeTransfer',
      },
    ]);
    expect(links).toMatchSnapshot();
  });

  it('should throw an error if arguments are not provided', async () => {
    mocks.request.send.mockReturnValue(() =>
      Promise.reject(new Error('Network error.'))
    );

    try {
      await addLinks();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });
});
