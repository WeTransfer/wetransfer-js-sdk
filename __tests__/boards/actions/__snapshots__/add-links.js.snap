// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add links action should create an add links request 1`] = `
Array [
  Object {
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
    "type": "link",
    "url": "https://wetransfer.com",
  },
]
`;

exports[`Add links action should throw an error if arguments are not provided 1`] = `[Error: There was an error when adding links to the board.]`;
